pkgs/development/compilers/julia/1.6.nix

```
@@ -0,0 +1,199 @@
+{
+fetchzip, fetchpatch, lib, pkgs, stdenv
+# Build tools
+, cmake, gfortran, openssl, perl, python3, which
+# Dependencies
+, cacert, curl, blas, gmp, lapack, libgit2, libnghttp2, libssh2, mbedtls, mpfr
+, p7zip, pcre2, zlib
+# Darwin
+, ApplicationServices, CoreServices
+}:
+
+# Ensure dependencies from Nixpkgs are equal or greater to vendored versions.
+assert lib.versionAtLeast (lib.strings.getVersion gmp) "6.2.0";
+assert lib.versionAtLeast curl.version "7.73.0";
+assert lib.versionAtLeast libgit2.version "1.1.0";
+assert lib.versionAtLeast libnghttp2.version "1.41.0";
+assert lib.versionAtLeast libssh2.version "1.9.0";
+# Nixpkgs is currently behind Julia here.
+#assert lib.versionAtLeast mbedtls.version "2.24.0";
+assert lib.versionAtLeast mpfr.version "4.1.0";
+assert lib.versionAtLeast p7zip.version "16.2.0";
+assert lib.versionAtLeast pcre2.version "10.35";
+assert lib.versionAtLeast zlib.version "1.2.11";
+# Can not assert for BLAS and LAPACK due to multiple providers, Julia uses:
+#   * OpenBLAS: 0.3.10
+#   * LAPACK: 3.9.0
+
+# Ensure BLAS and Lapack integer size compatibility.
+assert (blas.isILP64 == lapack.isILP64);
+
+let
+  # Needed to explicitly link OpenBLAS_jll to the shared library.
+  openblas = pkgs.openblas.override { blas64 = blas.isILP64; };
+
+in stdenv.mkDerivation rec {
+  pname = "julia";
+  version = "1.6.0";
+  src = fetchzip {
+    url = "https://github.com/JuliaLang/${pname}/releases/download/v${version}/${pname}-${version}-full.tar.gz";
+    sha256 = "14qhd0vp2y9c6126niz37vkz3n7skd6h565fai3khhd0li63vka6";
+  };
```

```suggestion
let
  # Needed to explicitly link OpenBLAS_jll to the shared library.
  openblas = pkgs.openblas.override { blas64 = blas.isILP64; };
  
  src_sha256 = "14qhd0vp2y9c6126niz37vkz3n7skd6h565fai3khhd0li63vka6";

in stdenv.mkDerivation rec {
  pname = "julia";
  version = "1.6.0";
  src = fetchzip {
    url = "https://github.com/JuliaLang/${pname}/releases/download/v${version}/${pname}-${version}-full.tar.gz";
    sha256 = src_sha256;
  };
```
so that update scripts work nicely